### DISCUSSION TITLE
Text filter operators for UUID columns

### DISCUSSION THREAD

**Discussion by JoshTheDerf** [directus/directus#16936] [Feature Requests]
Hi there,
We have a use-case where we'd like to be able to use `starts_with` filter conditions in the Directus App with UUID column types (Using the first N chars of a UUID as a "friendly" but less-unique id for staff.) Our database is currently SQLite.

Currently [only `eq`, `neq`, `null`, `nnull`, `in`, and `nin` are supported for UUID column types](https://github.com/directus/directus/blob/main/packages/shared/src/utils/get-filter-operators-for-type.ts#L43), I suspect because UUIDs are stored as `binary(16)` in PostgresSQL, and it doesn't support `LIKE` on binary columns without an additional cast.

Is there a way we could work around this, or fallback for Postgres to allow other databases to use this feature?

**Comment by MichaelFBA** [comment]
There is also the need to support `gt` `gte` `lt` `lte` for TEXT fields. Which should be supported IMO

**Reply by br41nslug** [reply]
Using numeric operators on textual data is indeed supported by most database vendors however as the implementations differ a lot between them, some will try and cast numeric strings and some wont or use different comparison approaches resulting in different results. We aim to offer a consistent experience across supported vendors and variety in implementation often makes it more confusing to use and hard to document.

**Reply by filiphanes** [reply]
When choosing between "consistently not working" and "working inconsistently depending on backend db" I would choose the latter in this case. In documentation you can write that exact ordering depends on backend db settings.

**Reply by filiphanes** [reply]
and by adding these operators on more fields you would add consistency between fields

**Comment by filiphanes** [comment]
In my use case, I need to use `gt` `gte` `lt` `lte` for UUID fields to avoid using inefficient queries.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
